asv_bench/benchmarks/benchmarks.py

```diff
@@ -819,4 +819,185 @@ def time_index(self, shape):
+        self.s2 = IMPL.Series(np.random.randn(len(mi)), index=mi)
+        self.s2_subset = self.s2[::2].copy()
+        execute(self.df)
+        execute(self.df2)
+        execute(self.s)
+        execute(self.s2_subset)
+        execute(self.s2)
+        execute(self.s_subset)
+        execute(self.s_subset_no_cache)
+
+    def time_reindex_dates(self, shape):
+        execute(self.df.reindex(self.rng_subset))
+
+    def time_reindex_columns(self, shape):
+        execute(self.df2.reindex(columns=self.df.columns[1:5]))
+
+    def time_reindex_multiindex_with_cache(self, shape):
+        # MultiIndex._values gets cached (pandas specific)
+        execute(self.s.reindex(self.s_subset.index))
+
+    def time_reindex_multiindex_no_cache(self, shape):
+        # Copy to avoid MultiIndex._values getting cached (pandas specific)
+        execute(self.s.reindex(self.s_subset_no_cache.index.copy()))
+
+    def time_reindex_multiindex_no_cache_dates(self, shape):
+        # Copy to avoid MultiIndex._values getting cached (pandas specific)
+        execute(self.s2_subset.reindex(self.s2.index.copy()))
+
+
+class TimeReindexMethod:
+
+    params = [
+        get_benchmark_shapes("TimeReindexMethod"),
+        ["pad", "backfill"],
+        [IMPL.date_range, IMPL.period_range],
+    ]
+    param_names = ["shape", "method", "constructor"]
+
+    def setup(self, shape, method, constructor):
+        N = shape[0]
+        self.idx = constructor("1/1/2000", periods=N, freq="1min")
+        self.ts = IMPL.Series(np.random.randn(N), index=self.idx)[::2]
+        execute(self.ts)
+
+    def time_reindex_method(self, shape, method, constructor):
+        execute(self.ts.reindex(self.idx, method=method))
+
+
+class TimeFillnaMethodSeries:
+
+    params = [get_benchmark_shapes("TimeFillnaMethodSeries"), ["pad", "backfill"]]
+    param_names = ["shape", "method"]
+
+    def setup(self, shape, method):
+        N = shape[0]
+        self.idx = IMPL.date_range("1/1/2000", periods=N, freq="1min")
+        ts = IMPL.Series(np.random.randn(N), index=self.idx)[::2]
+        self.ts_reindexed = ts.reindex(self.idx)
+        self.ts_float32 = self.ts_reindexed.astype("float32")
+        execute(self.ts_reindexed)
+        execute(self.ts_float32)
+
+    def time_reindexed(self, shape, method):
+        execute(self.ts_reindexed.fillna(method=method))
+
+    def time_float_32(self, shape, method):
+        execute(self.ts_float32.fillna(method=method))
+
+
+class TimeFillnaMethodDataframe:
+
+    params = [get_benchmark_shapes("TimeFillnaMethodDataframe"), ["pad", "backfill"]]
+    param_names = ["shape", "method"]
+
+    def setup(self, shape, method):
+        N = shape[0]
+        self.idx = IMPL.date_range("1/1/2000", periods=N, freq="1min")
+        td = IMPL.DataFrame(np.random.randn(*shape), index=self.idx)[::2]
+        self.td_reindexed = td.reindex(self.idx)
+        self.td_float32 = self.td_reindexed.astype("float32")
```

Comment:
I wonder why cast types to `float32`, if after `np.random.randn` function it should already be float numbers.

Reply:
The np.random.randn returns data in float64. In pandas ASV benchmarks for pd.Series benchmarking was done for float64 and float32. So type casting was done to convert from fp64 to float32.

Reply:
ok, let's see if there is any difference between these two types for Modin.